src/protagonist/Orchestrator/Infrastructure/Auth/Auth2AccessValidator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DLCS.Core.Types;
+using Orchestrator.Infrastructure.Auth.V2;
+
+namespace Orchestrator.Infrastructure.Auth;
+
+/// <summary>
+/// <see cref="IAssetAccessValidator"/> that uses external service to validate access via IIIF Auth v2
+/// </summary>
+public class Auth2AccessValidator : IAssetAccessValidator
+{
+    private readonly IIIFAuth2Client iiifAuth2Client;
+
+    public Auth2AccessValidator(IIIFAuth2Client iiifAuth2Client)
+    {
+        this.iiifAuth2Client = iiifAuth2Client;
+    }
+    
+    public async Task<AssetAccessResult> TryValidate(AssetId assetId, List<string> roles, AuthMechanism mechanism,
+        CancellationToken cancellationToken = default)
+    {
+        // NOTE(DG) - caller of this has checked appropriate cookie exists
```

Comment:
Yes, I left it in there as it's not obvious, when `AuthMechanism` can be `BearerToken` or `Cookie`, I'm aborting if `mechanism == AuthMechanism.BearerToken`.

The current DLCS has a single `IAssetAccessValidator` impl, which supports legacy IIIF Auth - 0.9 and 1 (this _was_ `AssetAccessValidator` but renamed to `Auth1AccessValidator`). This PR introduces 2 more `IAssetAccessValidator` implementations:

* `Auth2AccessValidator` - this support IIIF Auth2 and proxies to [iiif-auth-v2](https://github.com/dlcs/iiif-auth-v2)
* `AssetAccessValidator` - since we support Auth0/1 + Auth2, this is the class that is injected into consuming services and works out how to authorize requests - either using `Auth1AccessValidator` or `Auth2AccessValidator`. The only way that `Auth2AccessValidator` will be called is via `AssetAccessValidator`, which has checked the relevant cookie exists hence the comment.

`Auth1AccessValidator` and `AssetAccessValidator` are temporary implementations, only intended to bridge us over while we deploy Auth2. Once Auth2 is out we are removing support for Auth0/1. At that point the cookie check will be moved back into `Auth2AccessValidator` and that comment can go.